src/page/mixin/cartridge-behaviour.js

```diff
@@ -25,14 +23,20 @@ module.exports = {
                 }
             `);
         }
-        
-        setHeader(this.cartridgeConfiguration());
+        const config = hasCartridge ? cartridgeConfiguration() : {};
+        setHeader(config);
     },
 
     /**
      * Registers the cartridge upon mounting.
      */
     componentWillMount() {
         this._registerCartridge();
+    },
+
+    componentWillReceiveProps(nextProps) {
```

Comment:
Please comment your functions next time (I think you can inherit doc from React for lifecycle, using @inheritdoc)